src/System.Threading.Channels/tests/Stress.cs

```diff
@@ -0,0 +1,144 @@
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System;
+using Xunit;
+
+namespace System.Threading.Channels.Tests
+{
+    public class StressTests
+    {
+        public static IEnumerable<object[]> TestData()
+        {
+            foreach (bool singleReader in new [] {false, true})
+                foreach (bool singleWriter in new [] {false, true})
+                    foreach (bool allowSynchronousContinuations in new [] {false, true})
+                    {
+                        Func<ChannelOptions, Channel<int>> unbounded = o => Channel.CreateUnbounded<int>((UnboundedChannelOptions)o);
+                        yield return new object[] { unbounded, new UnboundedChannelOptions
+                                                                    {
+                                                                        SingleReader = singleReader,
+                                                                        SingleWriter = singleWriter,
+                                                                        AllowSynchronousContinuations = allowSynchronousContinuations
+                                                                    }};
+                    }
+
+            for (BoundedChannelFullMode bco = BoundedChannelFullMode.Wait;  bco <= BoundedChannelFullMode.DropWrite; bco++)
+                foreach (int capacity in new [] { 1, 1000 })
+                    foreach (bool singleReader in new [] {false, true})
+                        foreach (bool singleWriter in new [] {false, true})
+                            foreach (bool allowSynchronousContinuations in new [] {false, true})
+                            {
+                                Func<ChannelOptions, Channel<int>> bounded = o => Channel.CreateBounded<int>((BoundedChannelOptions)o);
+                                yield return new object[] { bounded, new BoundedChannelOptions(capacity)
+                                                                        {
+                                                                            SingleReader = singleReader,
+                                                                            SingleWriter = singleWriter,
+                                                                            AllowSynchronousContinuations = allowSynchronousContinuations,
+                                                                            FullMode = bco
+                                                                        }};
+                            }
+
+        }
+
+        const int MaxNumberToWriteToChannel = 1_000_000;
+        private static readonly int MaxTaskCounts = Math.Max(2, Environment.ProcessorCount);
+
+        [OuterLoop]
+        [Theory]
+        [MemberData(nameof(TestData))]
+        public void RunInStressMode(Func<ChannelOptions, Channel<int>> channelCreator, ChannelOptions options)
```

Comment:
Is this meant to be the only stress test?  I would have expected different tests for different things.  For example, this doesn't test WaitForRead/WriteAsync, it doesn't test TryWrite, and it likely doesn't exercise various patterns of member X being used concurrently with member Y.  It also doesn't stress completion, how it interacts with concurrent attempts to read/write, etc.

The test should be named for the thing that it's trying to stress, and ideally there would be tests for all interesting aspects.

That then contributes to my concern about this being [Outerloop].  Even if this test takes 1 minute (which I still think is too long for Outerloop... imagine if every project had an additional 1 minute test, that could add hours to a corefx run), if each such test took a minute we'd be at much more than that.

This test seems like a reasonable stressing of WriteAsync + TryRead/ReadAsync, although even then I'm not sure it's really going to stress WriteAsync+ReadAsync in a way that might be desired, but a lot of the reads will be satisfied by TryRead.

Reply:
- Currently we don't have the infrastructure in corefx for stress tests. I already talked before with @weshaggard and he suggested to keep such test as outerloop. if you have a better suggestion, please let me know and I'll be happy to do it.

- I'll add another test to ensure exercising WriteAsync and ReadAsync (with using WaitForRead/Write.

Reply:
> Currently we don't have the infrastructure in corefx for stress tests. I already talked before with @weshaggard and he suggested to keep such test as outerloop

@karelz, is this how we want to handle stress tests?